docs/proposals/ThumbnailGallery/README.md

```diff
@@ -0,0 +1,60 @@
+# Component Name
+
+The `ThumbnailGallery` component will accept a list of attachments for images
+and display them as thumbnails. This would also have an option to show the
+images in a LightBox component on click of those images.
+
+## Accessibility
+
+Assistive technology should be able to see the images and know the image name
+based on the `name` property of each attachment.
+
+## Responsiveness & Mobile
+
+This component show maximum number of images possible in a line and show the
+count of images left to show.
+
+## Wireframe
+
+[mockup](mockup.png)
+
+## Interface
+
+```
+<ThumbnailGallery
+    attachments={[{
+            name: "filename.png",
+            url: "https://url-to-that-image"
+            type: "image/png"
+            size: 12345
+`   }]}
+    openGalleryOnClick=true|false
+/>
+```
+
+Internally ThumbnailGallery would look like this along with some style details
+
+```
+{attachemnts && attachments.map(attachment => {
+    return (<FormatFile file={{
```

Comment:
I think we may want to use a `FormatImage` instead of a `FormatFile` here instead. `FormatImage` unfortunately doesn't yet exist. I know @JeffMarvin is looking at doing this for Hacktoberfest. I'm not sure where he's at with that. In lue of that though we could make a quick internal version specifically for the `Gallery`…

Reply:
We are currently exploring something that could combine both an uploaded fie/image or a URL if that is known. Part of the implementation will try to work with our current types to create something flexible and clear.